### PR TITLE
HTBHF-2049 Added address lookup feature toggle.

### PR DESCRIPTION
**Pull request by dwybourn**
- Added feature toggle support. Either postcode or manual address page will be run depending on an environment variable feature toggle.
- CD build script is to be updated to set this environment variable.

### CODE REVIEW

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala [hunk 1]
```diff
@@ -44,6 +49,24 @@ class ClaimSimulation extends Simulation {
     randomChar.toString
   }
 
+  private def addressStep =
+    if (features.getOrElse("ADDRESS_LOOKUP", false)) {
```
Comment (mikeSmith08) [inline src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: Weren't we using the toggle format of "ADDRESS_LOOKUP_ENABLED"?

File: src/main/scala/uk/gov/dhsc/htbhf/RunGatlingTests.scala.tmp [hunk 2]
```diff
@@ -19,7 +19,7 @@ object RunGatlingTests {
     try {
       val result = Gatling.fromMap(config)
       // write the result to system.out so we can identify success or failure from the logs
-      System.out.println("Finished running gatling tests - result=" + result)
+      System.out.println("Finished running gatling tests with feature toggles: " + sys.env.get("FEATURE_TOGGLES") + " - result=" + result)
```
Comment (YetAnotherMatt) [inline src/main/scala/uk/gov/dhsc/htbhf/RunGatlingTests.scala.tmp]: Can we log the feature toggles on a separate line? There could eventually be quite a few of them. Also, the CD scripts look for this _exact_ string: "Finished running gatling tests - result="

Comment (YetAnotherMatt) [reply src/main/scala/uk/gov/dhsc/htbhf/RunGatlingTests.scala.tmp]: Actually, while we're there, we should probably add a comment to that effect

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala (hunk 1 above)
Comment (dwybourn) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: Fixed.

File: src/main/scala/uk/gov/dhsc/htbhf/RunGatlingTests.scala.tmp (hunk 2 above)
Comment (dwybourn) [reply src/main/scala/uk/gov/dhsc/htbhf/RunGatlingTests.scala.tmp]: Added log to it's own line and added comment about the line not too edit.

File: src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala [hunk 3]
```diff
@@ -44,6 +47,24 @@ class ClaimSimulation extends Simulation {
...
 
+  private def addressStep =
+    if (features.getOrElse("ADDRESS_LOOKUP_ENABLED", false)) {
+      http("send postcode")
+        .post("/postcode")
+        .formParam("postcode", "AA1 1AA")
+        .formParam("_csrf", "${csrf_token2}")
+    }
+    else {
+      http("send_manual_address")
```
Comment (mikeSmith08) [inline src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: Won't the manual address page always be present, so if their address cannot be found by the postcode lookup then they'll go to the manual address page? If so, I think this should be outside the else block and we only have the ADDRESS_LOOKUP_ENABLED toggle on the postcode lookup page.

Comment (dwybourn) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: Updated so that postcode and manual address are entered if the lookup is enabled.

Comment (dwybourn) [reply src/main/scala/uk/gov/dhsc/htbhf/ClaimSimulation.scala]: Updated to go to manual address as well as postcode if enabled.

**Pull request merged**